cmake/config_validation.cmake

```
@@ -1,71 +1,138 @@
-# Name: config_validation.cmake file
-# ThundeRatz Robotics Team
-# Brief: This file contains the existence checks of the declared variables
-# 04/2023
-
 ###############################################################################
-## Auxiliary Sets
+## Existence checks
 ###############################################################################
 
-# This variable is used by the stm32-cmake lib to find the STM32CubeMX files
-# @see: https://github.com/ObKo/stm32-cmake#configuration
-set(STM32_CUBE_${DEVICE_CORTEX}_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cube)
-
-# This set contains all the variables that must be defined by the user
-# It is used to check if all of them are properly defined
-set(USER_INPUT_VARIABLES
-    DEVICE_CORTEX
-    DEVICE_FAMILY
-    DEVICE_TYPE
-    DEVICE_DEF
-    DEVICE_FAMILY_COMPACT
-    DEVICE
-    BOARD_VERSION
-    TARGET_BOARD
-)
+# Check if CMake build type is correctly configured
+if(NOT (BUILD_TYPE STREQUAL "Release"        OR BUILD_TYPE STREQUAL "Debug" OR
+        BUILD_TYPE STREQUAL "RelWithDebInfo" OR BUILD_TYPE STREQUAL "MinSizeRel"))
+    set(BUILD_TYPE "RelWithDebInfo")
+endif()
 
-###############################################################################
-## Existence checks
-###############################################################################
+set(CMAKE_BUILD_TYPE ${BUILD_TYPE})
+message(STATUS "Build type: ${CMAKE_BUILD_TYPE}")
 
-## Check if STM32CubeMX variables are properly defined
-if(DEFINED ENV{CUBE_PATH})
-    message(STATUS "CUBE_PATH defined as $ENV{CUBE_PATH}")
-else()
-    message(FATAL_ERROR "CUBE_PATH not defined")
+# Check if host system is Linux
+if(NOT CMAKE_HOST_SYSTEM_NAME STREQUAL "Linux")
```

```suggestion
if(NOT CMAKE_HOST_UNIX)
```
Acho que se deixar assim ele suporta MacOS também.